test/integration/default/controls/_mapdata_spec.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+# Replace per minion strings
+replacement = {
+  hostname: system.hostname
+}
+
+# Keep only first 2 digits from Ubuntu finger
+mapdata_file = "_mapdata/#{system.platform[:finger].split('.').first}.yaml"
+
+# Load the mapdata from profile https://docs.chef.io/inspec/profiles/#profile-files
+mapdata_dump = inspec.profile.file(mapdata_file) % replacement
```

Comment:
Just starting a conversation here but shouldn't we keep these in the narrowest possible scope, i.e. moving these into the `control` scope?

Reply:
Ho yes, you did that for `openntpd`. I don't know if there is an InSpec best practices for this but I personaly try to avoid doing much logic in the `control/end` block like I avoid the same in SaltStack states block ;-)